### ISSUE TITLE
Support new SDS version Archive 21

### ISSUE THREAD

**Issue by asadboi123** [Greavesy1899/MafiaToolkit#90]
TopSpin 2K25 from Hangar 13 also uses the same engine and SDS files, but the version it's using is newer so I'm getting the error "Unsupported Archive Version 21"

Is it possible to support it? Might help for Mafia 4 if they use same one. I can provide game files if needed

**Comment by Greavesy1899** [comment]
Some of us are exploring the idea of adding support, but it appears that the developers have chosen to add some runtime checks to avoid tampering. We've got enough to unpack the files, but not repack, as we would need to overcome this anti-tamper check anyway.

I think if they continue to use the engine for Mafia IV, we are okay with the knowledge we have of version 21. But to be fair, at that point, if the developers are pushing for anti-tamper techniques then it does make me begin to wonder whether we should make mods for their games anyway. That's a debate for when the new game is released though.

TLDR: We know the changes in version 21 and can unpack them, but cannot repack them or even test them due to these runtime checks.

**Comment by asadboi123** [comment]
Ah, good to know you guys are already on top of it. Do you think you could add just the unpacking changes, maybe in a separate branch? I'm also thinking they only added runtime verification checks due it to it being a cross-play sports title. Would make little sense to add that in mafia 4 imo but who knows, it's 2k at the end of the day

**Comment by Greavesy1899** [comment]
I can see about making unpacking live on a separate branch
